### PR TITLE
lancedb integration improvements

### PR DESCRIPTION
**Pull request by AyushExel**
- **Description:** 
  Hi, I'm from lancedb team.
 - Improves lancedb integration by making it easier to use - now you don't require to create tables manually and pass it in the constructor, although that is still backwards compatible.
 - Bug fix - pandas was being used even though its not a dependency for lancedb or langchain

  - **Tag maintainer:** 
  @baskaryan

  - **Twitter handle:** @loldedxd

### CODE REVIEW

**Comment by AyushExel** [comment]
@baskaryan Fixed linting. Please take a look again, thanks!

**Comment by AyushExel** [comment]
hi @baskaryan @nfcampos @hwchase17 can I get a review on this? Thanks!

File: libs/langchain/langchain/vectorstores/lancedb.py [hunk 1]
```diff
@@ -134,3 +165,24 @@ def from_texts(
         instance.add_texts(texts, metadatas=metadatas, **kwargs)
 
         return instance
+
+    def _init_table(self) -> Any:
+        import lancedb
```
[the remark this answers is not in the document]
Comment (AyushExel) [reply libs/langchain/langchain/vectorstores/lancedb.py]: @baskaryan The first import happens in the contructor which is handled by try/except here https://github.com/langchain-ai/langchain/pull/11858/files#diff-c2377771bdd8f055bd908efed9a8f1362ceacce7b6881b0f29b48cfedd82e323R52

File: libs/langchain/langchain/vectorstores/lancedb.py [hunk 2]
```diff
@@ -25,33 +39,42 @@ class LanceDB(VectorStore):
...
-            )
-        self._connection = connection
         self._embedding = embedding
         self._vector_key = vector_key
         self._id_key = id_key
         self._text_key = text_key
+        self._table_name = table_name
+
+        if self._embedding is None:
+            self._embedding = OpenAIEmbeddings()
```
[the remark this answers is not in the document]
Comment (AyushExel) [reply libs/langchain/langchain/vectorstores/lancedb.py]: Done

**Comment by AyushExel** [comment]
@baskaryan addressed the reviews. Thanks for reviewing

**Comment by AyushExel** [comment]
Hi @baskaryan can you please take a look again? Thanks!

**Comment by AyushExel** [comment]
@baskaryan friendly ping :)

**Comment by AyushExel** [comment]
> hi @baskaryan @nfcampos @hwchase17 can I get a review on this? Thanks!

**Comment by AyushExel** [comment]
@jeremi I haven't been able to get any traction on this PR. Can you please take a look? It is very important usage enhancement for our users. Thanks!

**Comment by e-ave** [comment]
@baskaryan I would also like to see this merged

**Pull request closed without merging**